Stage_2

KinCadeFoster opened the pull request:
- Инициализирован массив dictionary
- На каждой итерации прохода по строкам файла создаём объект типа Word и добавляем его в dictionary
- Добавил информацию о количестве правильных ответов в файл words.txt и в сущность Word
- Распечатал содержимое dictionary в консоль
- Добавил обработку ошибок прасинга строк, что бы мы могли обработать валидные строки при возникновении ошибок

KinCadeFoster committed:
Stage_2
- Инициализирован массив dictionary
- На каждой итерации прохода по строкам файла создаём объект типа Word и добавляем его в dictionary
- Добавил информацию о количестве правильных ответов в файл words.txt и в сущность Word
- Распечатал содержимое dictionary в консоль
- Добавил обработку ошибок прасинга строк, что бы мы могли обработать валидные строки при возникновении ошибок

danelyan commented on src/main/kotlin/Main.kt:
```diff
@@ -1,8 +1,24 @@
+        try {
+            val lineSplit = string.split(" | ")
+            dictionary.add(Word(enWord = lineSplit[0], ruWord = lineSplit[1], correctCount = lineSplit[2].toInt()))
+        } catch (e: Exception) {
```
давай заменим try..catch {}:
1. Будем уточнять размер `lineSplit`
2. Будем использовать `toIntOrNull() ?: 0` вместо `toInt()`, чтобы не падать если там окажется не число, а строка

danelyan replied:
И раз у тебя в файле есть пробелы, давай делать trim() для строк

KinCadeFoster committed:
Stage_2
- Добавил проверку размера lineSplit
- добавил toIntOrNull() ?: 0 что бы быть уверенным что там число
- добавил trim() для обработки слов

danelyan reviewed (approved):
Отлично!
